src/main/java/catan/services/impl/PlayServiceImpl.java

```diff
@@ -38,10 +42,12 @@
     public static final String ERROR_CODE_ERROR = "ERROR";
 
     private GameDao gameDao;
+    private RandomUtil randomUtil;
     private GameUtil gameUtil;
     private PlayUtil playUtil;
     private BuildUtil buildUtil;
     private PreparationStageUtil preparationStageUtil;
+    private MainStageUtil mainStageUtil;
```

Comment:
since we inject preparationStageUtil, mainStageUtil and playUtil to playService and also preparationStageUtil and mainStageUtil to playUtil , I think we have wrong architecture, we should think how to refactor it